web/src/components/countryhistorypricesgraph.js

```diff
@@ -0,0 +1,129 @@
+} from '../helpers/history';
+
+import AreaGraph from './graph/areagraph';
+
+const prepareGraphData = (historyData, colorBlindModeEnabled, electricityMixMode) => {
+  if (!historyData || !historyData[0]) return {};
+
+  const currencySymbol = getSymbolFromCurrency(((first(historyData) || {}).price || {}).currency);
+  const valueAxisLabel = `${currencySymbol || '?'} / MWh`;
+
+  const priceMaxValue = d3Max(historyData.map(d => (d.price || {}).value));
+  const priceColorScale = scaleLinear()
+    .domain([0, priceMaxValue])
+    .range(['yellow', 'red']);
+
+  const data = historyData.map(d => ({
+    [PRICES_GRAPH_LAYER_KEY]: d.price && d.price.value,
+    datetime: moment(d.stateDatetime).toDate(),
+    // Keep a pointer to original data
+    _countryData: d,
+  }));
+
+  const layerKeys = [PRICES_GRAPH_LAYER_KEY];
+  const layerStroke = () => 'darkgray';
+  const layerFill = () => '#616161';
+  const focusFill = key => d => priceColorScale(d.data[key]);
+
+  return {
+    data,
+    layerKeys,
+    layerStroke,
+    layerFill,
+    focusFill,
+    valueAxisLabel,
+  };
+};
+
+const mapStateToProps = state => ({
+  colorBlindModeEnabled: state.application.colorBlindModeEnabled,
+  electricityMixMode: state.application.electricityMixMode,
+  startTime: getZoneHistoryGraphStartTime(state),
+  endTime: getZoneHistoryGraphEndTime(state),
+  historyData: getSelectedZoneHistory(state),
+  isMobile: state.application.isMobile,
+  selectedTimeIndex: state.application.selectedZoneTimeIndex,
+});
+
+const CountryHistoryPricesGraph = ({
+  colorBlindModeEnabled,
+  electricityMixMode,
+  startTime,
+  endTime,
+  historyData,
+  isMobile,
+  selectedTimeIndex,
+}) => {
+  const [selectedLayerIndex, setSelectedLayerIndex] = useState(null);
+
+  // Recalculate graph data only when the history data is changed
+  const {
+    data,
+    layerKeys,
+    layerStroke,
+    layerFill,
+    focusFill,
+    valueAxisLabel,
+  } = useMemo(
+    () => prepareGraphData(historyData, colorBlindModeEnabled, electricityMixMode),
+    [historyData, colorBlindModeEnabled, electricityMixMode]
+  );
+
+  // Mouse action handlers
+  const backgroundMouseMoveHandler = useMemo(
+    () => createSingleLayerGraphBackgroundMouseMoveHandler(isMobile, setSelectedLayerIndex),
+    [isMobile, setSelectedLayerIndex]
+  );
+  const backgroundMouseOutHandler = useMemo(
+    () => createSingleLayerGraphBackgroundMouseOutHandler(setSelectedLayerIndex),
+    [setSelectedLayerIndex]
+  );
+  const layerMouseMoveHandler = useMemo(
+    () => createGraphLayerMouseMoveHandler(isMobile, setSelectedLayerIndex),
+    [isMobile, setSelectedLayerIndex]
+  );
+  const layerMouseOutHandler = useMemo(
+    () => createGraphLayerMouseOutHandler(setSelectedLayerIndex),
+    [setSelectedLayerIndex]
+  );
+
+  return (
+    <AreaGraph
+      data={data}
+      layerKeys={layerKeys}
+      layerStroke={layerStroke}
+      layerFill={layerFill}
+      focusFill={focusFill}
```

Comment:
what about calling this markerFill?

Reply:
I like that, it does sound more descriptive :) Addressed in https://github.com/tmrowco/electricitymap-contrib/pull/2254/commits/548ef41a2d9a2ec5447c38a057f5e1dc880d6f28.